index.js

```diff
@@ -53,7 +54,12 @@ function NYC (opts) {
   // require extensions can be provided as config in package.json.
   this.require = arrify(config.require || opts.require)
 
-  this.transform = this._createTransform()
+  this.extensions = arrify(config.extensions || opts.extension)
+
+  this.transform = this.extensions.concat(['.js']).reduce(function (transform, ext) {
+    transform[ext] = this._createTransform(ext)
```

Comment:
Perhaps lowercase `ext` here to make the later comparison easier.

I'm assuming `require.extensions` is case insensitive. If it isn't, well that would be weird.